src/cli/commands/beacon.ts

```diff
@@ -4,137 +4,91 @@
 
 import {CliCommand} from "./interface";
 import {CommanderStatic} from "commander";
-import {isPlainObject} from "../../util/objects";
-import  {LogLevel, WinstonLogger} from "../../logger";
-import BeaconNode, {BeaconNodeCtx} from "../../node";
-import {ethers} from "ethers";
-import {CliError} from "../error";
-import {IApiConstructor} from "../../rpc/api/interface";
-import * as RPCApis from "../../rpc/api";
+import {ILogger, LogLevel, WinstonLogger} from "../../logger";
+import {BeaconNode} from "../../node";
+import {BeaconNodeOptions, IBeaconNodeOptions} from "../../node/options";
+import {generateCommanderOptions, optionsToConfig} from "../util";
 import deepmerge from "deepmerge";
-import {getTomlConfig, IConfigFile} from "../../util/file";
-import defaults from "../../node/defaults";
-import {ILogger} from "../../logger";
-import {promptPassword} from "../../util/io";
-import fs from "fs";
+import {getTomlConfig} from "../../util/file";
+import Validator from "../../validator";
+import {RpcClientOverInstance} from "../../validator/rpc";
+import {BeaconApi, ValidatorApi} from "../../rpc";
 
 interface IBeaconCommandOptions {
-  db: string;
-  depositContract: string;
-  eth1RpcUrl: string;
-  rpc: string;
-  configFile: string;
+  configFile?: string;
   loggingLevel?: string;
-  validator?: {
-    key: string;
-    db?: string;
-  };
+  [key: string]: string;
```

Comment:
what is this?

Reply:
it allows any key value property so we dont have to define generated options again here. it gets validate later so everything unsupported gets remove

Reply:
why not just use a map?

Reply:
I wan't to have configFile and loggingLevel props typed